### PR TITLE
feat(next): add subplat glean server yaml

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Define glean events and metrics for subplat backend events

## This pull request

- Adds metrics and events to subplat backend metrics yaml

## Issue that this pull request solves

Closes: #FXA-10086

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 1]
```diff
@@ -1,3 +1,390 @@
...
+      - subplat-team@mozilla.com
+    bugs:
+      - https://mozilla-hub.atlassian.net/browse/FXA-7531
+    data_reviews:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=tbd
+    expires: never
+    data_sensitivity:
+      - interaction
+  fail:
+    description: Checkout fail event
```
Comment (LZoog) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:380]: Quick drive-by comment - for some of our events like failures, FxA adds a 'reason' ([see example](https://github.com/mozilla/fxa/blob/f39d66b07fac16ea34875d061b10192ec5fd4e8a/packages/fxa-settings/src/pages/Signin/utils.ts#L172)). Not sure if you want this too.

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml [hunk 2]
```diff
@@ -1,3 +1,390 @@
 ---
 # Schema
 $schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
+
+account:
+  user_id_sha256:
+    type: string
+    description: |
+      A hex string of a sha256 hash of the account's uid
```
Comment (chenba) [inline libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:9]: Perhaps chat with Chelsea first, but we've been cleared to use the plain FxA uid now.

File: libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml (hunk 1 above)
Comment (StaberindeZA) [reply libs/shared/metrics/glean/src/registry/subplat-backend-metrics.yaml:380]: As part of the SubPlat P1 Metrics epic from last year, the `reason` field was actually renamed to `error_id` for the SubPlat events. Not sure exactly what the motivation was for that change, but I figured we'd stick with that. [SubPlat P1 Product Metrics - FxA Epic Technical Specification](https://docs.google.com/document/d/1-KdXzZbn3Utw2aekJbcc1lrMV9tNb19tzifPbdtQaso/edit)

**Comment by chenba** [comment]
Presumably `send_in_pings` has been intentionally left out the YAML?

**Comment by StaberindeZA** [comment]
> Presumably `send_in_pings` has been intentionally left out the YAML?

Yup, I figured the default would work, since events would be sent in the `events` ping by default? ([link to docs](https://mozilla.github.io/glean/book/reference/yaml/metrics.html#send_in_pings))

**Review by julianpoy — APPROVED** [review]

**Pull request merged**